pontoon/base/static/js/translate.js

```diff
@@ -4015,7 +4015,16 @@ var Pontoon = (function (my) {
         return null;
       }
 
-      return decodeURIComponent(results[2].replace(/\+/g, " "));
+      var encodedURI = results[2].replace(/\+/g, " ");
+
+      try {
+        return decodeURIComponent(encodedURI);
+
+      // If querystring not encoded, we need to encode it first
+      } catch (e if e instanceof URIError) {
```

Comment:
This inline conditional catch syntax [is not standard and its use is discouraged](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#Conditional_catch_clauses). You should check for `URIError` in a separate conditional block.

Reply:
Thanks. I had no idea. And I actually found this syntax in the exact same article. :/